e2e/rbd.go

```diff
@@ -427,6 +428,136 @@ var _ = Describe("RBD", func() {
 				validateRBDImageCount(f, 0)
 			})
 
+			By("restart rbd-nbd process on nodeplugin and continue IO after nodeplugin restart", func() {
+				err := deleteResource(rbdExamplePath + "storageclass.yaml")
+				if err != nil {
+					e2elog.Failf("failed to delete storageclass with error %v", err)
+				}
+				// Tweak Storageclass to add netlink,reattach rbd-nbd mounter options
+				scOpts := map[string]string{
+					"mounter":    "rbd-nbd",
+					"mapOptions": "try-netlink,reattach-timeout=180",
+				}
+				err = createRBDStorageClass(f.ClientSet, f, nil, scOpts, deletePolicy)
+				if err != nil {
+					e2elog.Failf("failed to create storageclass with error %v", err)
+				}
+
+				pvc, err := loadPVC(pvcPath)
+				if err != nil {
+					e2elog.Failf("failed to load PVC with error %v", err)
+				}
+				pvc.Namespace = f.UniqueName
+
+				app, err := loadApp(appPath)
+				if err != nil {
+					e2elog.Failf("failed to load application with error %v", err)
+				}
+				app.Namespace = f.UniqueName
+				label := map[string]string{
+					"app": app.Name,
+				}
+				app.Labels = label
+				app.Spec.Volumes[0].PersistentVolumeClaim.ClaimName = pvc.Name
+				err = createPVCAndApp("", f, pvc, app, deployTimeout)
+				if err != nil {
+					e2elog.Failf("failed to create PVC and application with error %v", err)
+				}
+				// validate created backend rbd images
+				validateRBDImageCount(f, 1)
+
+				// Get the label name of daemonset dynamically, as the label name is not same for helm and non-helm deploys
+				ds, err := f.ClientSet.AppsV1().DaemonSets(cephCSINamespace).Get(context.TODO(), rbdDaemonsetName, metav1.GetOptions{})
```

Comment:
@nixpanic @Madhu-1 JFYI - this way I'm now able to get the label dynamically.

Reply:
Thanks @pkalever

Reply:
Use this label selector to delete the daemonset pods also?

Reply:
New line number 509?
I'm doing the same right?
'm I missing anything?

Reply:
i was referring to https://github.com/ceph/ceph-csi/pull/1839/files#diff-bfe7fe7ada8a43a75c21ddfd9535e10f37d6e55e9a3006ce5a1872f73f3ae198R399-R402

Reply:
Hmm that was part of a different test scenario. But it's a good idea to keep this logic in the utils and reuse it where ever needed. I will fix it, thanks!

Reply:
@Madhu-1 done now. Thanks!